ir/instr.cpp

```diff
@@ -1949,11 +1952,17 @@ StateValue ICmp::toSMT(State &s) const {
     fn = [this, &s, fn](const expr &av, const expr &bv, Cond cond) {
       Pointer lhs(s.getMemory(), av);
       Pointer rhs(s.getMemory(), bv);
-      if (use_provenance) {
+      switch (pcmode) {
+      case INTEGRAL:
+        return fn(lhs.getAddress(), rhs.getAddress(), cond);
+      case PROVENANCE:
         assert(cond == EQ || cond == NE);
         return cond == EQ ? lhs == rhs : lhs != rhs;
+      case OFFSETONLY:
+        return fn(lhs.getOffset(), rhs.getOffset(), cond);
+      default:
```

Comment:
setting a default in a switch of a enum is not a great idea as it elides future warnings of missing cases handling

Reply:
I moved it after the switch stmt.